src/uct/ib/mlx5/gga/gga_mlx5.c

```diff
@@ -609,22 +609,59 @@ static uct_iface_ops_t uct_gga_mlx5_iface_tl_ops = {
 };
 
 static int
-uct_gga_mlx5_iface_is_reachable_v2(const uct_iface_h tl_iface,
-                                   const uct_iface_is_reachable_params_t *params)
+uct_gga_mlx5_iface_is_same_device(const uct_iface_h tl_iface,
+                                  const uct_iface_addr_t *iface_addr,
+                                  const uct_iface_is_reachable_params_t *params)
```

Comment:
`uct_iface_is_reachable_params_t` looks redundant here, maybe just `const char* str, size_t length` and use `ucs_snprintf_safe`?

Reply:
if we drop `uct_iface_is_reachable_params_t ` then there will be code duplication:
```
    char *info_str_buf      = UCS_PARAM_VALUE(UCT_IFACE_IS_REACHABLE_FIELD,
                                              params, info_string, INFO_STRING,
                                              NULL);
    size_t info_str_buf_len = UCS_PARAM_VALUE(UCT_IFACE_IS_REACHABLE_FIELD,
                                              params, info_string_length,
                                              INFO_STRING_LENGTH, 0);
```

Reply:
We can to move most of logic of this function up, to uct_gga_mlx5_iface_is_reachable_v2().
So uct_gga_mlx5_iface_is_same_device() will get only tl_iface and iface_addr, and return 0/1. It will contain only the logic in lines 627-629. Then no need to define dummy params inside uct_gga_mlx5_ep_is_connected()

Reply:
there will still be code duplication:
```
    if (gga_addr == NULL) {
        uct_iface_fill_info_str_buf(params, "iface address is empty");
        return 0;
    }
```

Reply:
uct_iface_fill_info_str_buf(params, "iface address is empty"); should be only in uct_gga_mlx5_iface_is_reachable_v2; uct_gga_mlx5_ep_is_connected can just check if (gga_addr == NULL)

Reply:
`if (gga_addr == NULL)` will be the duplication

Reply:
IMO it's small enough (1 line) duplication and it can avoid extra initialization of dummy struct so it's worth it